streams/src/main/java/org/apache/kafka/streams/processor/TopologyBuilder.java

```diff
@@ -759,12 +848,49 @@ public synchronized ProcessorTopology build(Integer topicGroupId) {
         if (topicGroupId != null) {
             nodeGroup = nodeGroups().get(topicGroupId);
         } else {
-            // when nodeGroup is null, we build the full topology. this is used in some tests.
-            nodeGroup = null;
+            // when nodeGroup is null, we build the full topology minus the global groups
```

Comment:
comment does not apply anymore -- nodeGroup will never be null now (maybe empty though)